appengine/genetics/js/cage.js

```diff
@@ -403,128 +454,68 @@ Genetics.Cage.isMatingSuccessful = function(proposingMouse, askedMouse) {
     new Genetics.Cage.Event('MATE', proposingMouse.id, 'SELF').addToQueue();
     return false;
   }
-  if (proposingMouse.sex == askedMouse.sex) {
-   // If mice are not of different sex, mate does not succeed.
-    new Genetics.Cage.Event('MATE', proposingMouse.id, 'INCOMPATIBLE',
-        askedMouse.id).addToQueue();
-    return false;
-  }
-  if (askedMouse.fertility < 1) {
-    // If other mouse is infertile, mate does not succeed.
-    new Genetics.Cage.Event('MATE', proposingMouse.id, 'INFERTILE',
-        askedMouse.id).addToQueue();
-    return false;
-  }
   // Ask second mouse whether it will mate with the proposing mouse.
   var result = Genetics.Cage.runMouseFunction(askedMouse, 'acceptMate',
       proposingMouse);
   if (!result.success) {
     // Explode mouse if function threw an error or caused a timeout.
-    new Genetics.Cage.Event('MATE', proposingMouse.id, 'MATE_EXPLODED')
+    new Genetics.Cage.Event('MATE', proposingMouse.id, 'MATE_EXPLODED',
+        askedMouse.id)
+        .addToQueue();
+    new Genetics.Cage.Event('EXPLODE', askedMouse.id, 'acceptMate',
+        result.cause)
         .addToQueue();
+    Genetics.Cage.die(askedMouse);
     return false;
   }
   if (!result.value) {
-    // If mouse's response is positive.
+    // If mouse's response is negative.
     new Genetics.Cage.Event('MATE', proposingMouse.id, 'REJECTION',
         askedMouse.id).addToQueue();
     return false;
   }
+  // Asked mouse accepted the mating request.
+  // Use up one mating attempt for asked Mouse.
+  askedMouse.fertility--;
+  if (proposingMouse.sex == askedMouse.sex) {
+    // If mice are not of different sex, mate does not succeed.
+    new Genetics.Cage.Event('MATE', proposingMouse.id, 'INCOMPATIBLE',
+        askedMouse.id).addToQueue();
+    return false;
+  }
+  if (askedMouse.fertility < 0) {
+    // If other mouse is infertile, mate does not succeed.
+    new Genetics.Cage.Event('MATE', proposingMouse.id, 'INFERTILE',
+        askedMouse.id).addToQueue();
+    return false;
+  }
   return true;
 };
 
 /**
- * Adds the mouse to the cage.
+ * Adds the mouse to the cage and queues simulation for mouse.
  * @param {!Genetics.Mouse} mouse The mouse to add to the cage.
  */
-Genetics.Cage.born = function(mouse) {
-  // Adds child to population.
-  Genetics.Cage.aliveMice_.push(mouse);
+Genetics.Cage.addMouse = function(mouse) {
+  // Adds child to population and queue life simulation.
   Genetics.Cage.miceMap_[mouse.id] = mouse;
-  Genetics.Cage.life(mouse); // Queue life simulation for mouse.
+  Genetics.Cage.nextRoundMice_.unshift(mouse);
```

Comment:
Just to check, why unshift() instead of push()? Are mice processed from the end?